pitop/pma/button.py

```diff
@@ -18,6 +19,12 @@ def __init__(self, port_name, name="button"):
         self._pma_port = port_name
         self.name = name
 
+        if port_name not in Port.keys():
+            raise ValueError(f"{port_name} is not a valid port name. An example of a valid port name is D0")
+
+        if not port_name.startswith("D"):
+            raise ValueError(f"{port_name} is not a valid port type for a button. Try using a digital port, such as D0")
+
```

Comment:
I think that this should be handled by a mixin (see `Stateful`, `Recreatable`...) so that it doesn't need to be duplicated. Not sure what I'd call it though...

Reply:
Yeah I was thinking the same, I'd just call it "AnalogComponent" or "DigitalComponent". I wouldn't necessarily call it a "mixin" though, it's just standard inheritance (`Button` _is a_ `DigitalComponent`). Admittedly the name `DigitalComponent` does kind of suggest it does more than just port checking, so maybe "mixin" is the right terminology and it should be called DigitalPortValidator 😃

Reply:
Actually, `Button` inherits from gpiozero's `Button` class. It uses `Stateful` and `Recreatable` mixins are included, not inherited - the same would be the case for this?

Reply:
Well yeah but it's still inheritance - the only difference between a mixin and inheritance is generally just if the parent can't be used standalone it's called a "mixin". But then that definition doesn't really apply to abstract base classes, which is still inherited and definitely can't be used standalone. Anyway this is just terminology at this point, I think the idea makes sense